ml-dhs-aws/src/test/java/com/marklogic/client/dhs/test/TestJobRunner.java

```diff
@@ -37,11 +62,138 @@ public void testRun() throws IOException {
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobStream);
+			ObjectNode jobDef = (ObjectNode) jobControlObj.get("job");
+
+			String afterDocId = JobRunner.getAfterJobDocumentUri(jobDef.path("id").asText());
+			assertNotNull(jsonMgr.exists(afterDocId));
+			DocumentMetadataHandle afterDocumentMetadata = new DocumentMetadataHandle();
+			JacksonHandle readHandle = new JacksonHandle();
+			
+			assertNotNull(jsonMgr.read(afterDocId, afterDocumentMetadata, readHandle)); 
+			assertTrue(afterDocumentMetadata.getCollections().contains(JobRunner.getJobCollection(jobDef.path("id").asText())));
+			assertTrue(afterDocumentMetadata.getCollections().contains("/afterJob"));
+			assertTrue(afterDocumentMetadata.getPermissions().containsKey("admin"));
+			assertTrue(afterDocumentMetadata.getPermissions().get("admin").contains(Capability.READ));
+			assertTrue(afterDocumentMetadata.getPermissions().get("admin").contains(Capability.UPDATE));
+
+		} catch(Exception ex) {
+			ex.printStackTrace();
+		}
+	}
+	
+	@Test
+	public void testJsonDataStructure() throws Exception {
+		String csvFile = "data"+File.separator+"OrderLines.csv";
+
+		JobRunner jobRunner = new JobRunner();
+		String[] roles = { "admin" };
+		jobRunner.setRoles(roles);
+		JSONDocumentManager jsonMgr = client.newJSONDocumentManager();
+
+		try (InputStream csvStream = openStream(csvFile);
+				InputStream jobStream = openStream(JobRunner.jobFileFor(csvFile));) {
+			jobRunner.run(client, csvStream, jobStream);
+
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobStream);
+			ObjectNode jobDef = (ObjectNode) jobControlObj.get("job");
+
+			StructuredQueryBuilder queryBldr = new StructuredQueryBuilder();
+			// get only one result
+			jsonMgr.setPageLength(1);
+			
+			// match documents that are in the job collection other than the before or after document
+			DocumentPage docPage = jsonMgr.search(
+					queryBldr.andNot(queryBldr.collection(JobRunner.getJobCollection(jobDef.path("id").asText())),
+							queryBldr.document(JobRunner.getBeforeJobDocumentUri(jobDef.path("id").asText()),
+									JobRunner.getAfterJobDocumentUri(jobDef.path("id").asText()))),
+					1);
+			long docCount = docPage.getTotalSize();
+
+			FileReader fr = new FileReader(csvFile);
+			LineNumberReader lnr = new LineNumberReader(fr);
+			int linenumber = 0;
+			while (lnr.readLine() != null) {
+				linenumber++;
+			}
+			lnr.close();
+			// compare with the number of CSV records excluding the CSV header
+			assertEquals(docCount, (linenumber - 1));
+			
+			JacksonHandle firstDocHandle = docPage.nextContent(new JacksonHandle());
+			JsonNode firstObject = firstDocHandle.get();
+
+			// verify the record metadata and the instance keys by navigating the nodes using Jackson methods
+			assertNotNull(firstObject.get("metadata"));
+			assertNotNull(firstObject.get("instance"));
+
```

Comment:
Could the test verify the metadata properties?  (And the instance properties, based on the header line of the CSV file?)